tools/draw-importances.py

```diff
@@ -1,55 +1,180 @@
 #!/usr/bin/env python3
 import struct
 import sys
+import os
 from os.path import splitext
+import datetime
 import numpy as np
 from PIL import Image, ImageDraw
 from matplotlib import pyplot as plt
+import click
+import glob
+import re
 
+@click.command()
+@click.option('--path',  type = str,
+        help = "Absolute File locations of extracted hres.png and imps.bin ")
+@click.option('--input', help="Pass both hres.png and imps.bin",
+                default=[None] * 2, type=click.Tuple([str, str]))
 
-# Renders block importances output by ContextInner::compute_lookahead_data().
-# Usage:
-#        cargo run --features=dump_lookahead_data <input.y4m> -o /dev/null
-#        draw-importances.py <i-hres.png> <i-imps.bin>
-#        will output i-imps.png.
+@click.option('--verbose', is_flag=True, help="Will print verbose messages.")
+@click.option('--raw', is_flag=True, help="Print RAW Data of the bin.")
 
 
-with open(sys.argv[2], 'rb') as f:
-    contents = f.read()
+def blockImportance(input, verbose, path, raw):
+    """
+    CLI tool for extracting rav1e's Block Importance
 
-rows, cols = struct.unpack('qq', contents[:16])
-imps = np.frombuffer(contents[16:], dtype=np.float32).reshape((rows, cols))
+    Renders block importances output by ContextInner::compute_lookahead_data().
 
-# Use a fixed scale where anything >= 10 cannot be distinguished
-# to allow visually comparing multiple pictures
-max_imp = 10 ## Replace by `np.max(imps)` for relative scaling
+    Run rav1e binary with dump_lookahead_data feature flag to dump:\t\t
+    For eg: `cargo run --features=dump_lookahead_data <input.y4m> -o /dev/null`
 
-frame_size_multiplier = 4
+    Two modes available:\n
+    Mode 1: Simple file, passing one hres.png, and one imps.bin \t\t
+    Mode 2: Batch file processing for multiple hres and imps data\n
+    Output files will be saved in out/YYYYMMDD_HH folder in -imps.png files.
 
-frame = Image.open(sys.argv[1])
-frame = frame.resize((frame.width * frame_size_multiplier, frame.height * frame_size_multiplier))
-frame = frame.convert(mode='RGB')
+    """
+    if verbose:
+        click.echo("Input given: "+ str(input))
+        click.echo("Path given: " + str(path))
 
-mv_original_block_size = 8 // 2 # The importances are in 8×8 blocks, but we use half-resolution images.
-mv_block_size = mv_original_block_size * frame_size_multiplier
+    rx = re.compile(r'(\d\d-imps)|(\d-imps)')
```

Comment:
I'm not sure this is the best way of doing this (besides, this works only for 1 and 2 digits). Could you simply use something like `basename()` and `splitext()` to get the *-imps part of the name?

Reply:
indeed, makes sense and lot more easier